packages/aila/src/protocol/sectionToMarkdown.ts

```diff
@@ -40,9 +41,10 @@ export function sectionToMarkdown(
     try {
       const cycle = CycleOptionalSchema.parse(value);
       const content = `## ${cycle.title ?? "…"}\n\n`;
-      return `${content}\n\n### Explanation\n\n${sectionToMarkdown("explanation", cycle.explanation ?? "…")}\n\n### Check for understanding\n\n${cycle.checkForUnderstanding ? organiseAnswersAndDistractors(cycle.checkForUnderstanding) : "…"}\n\n### Practice\n\n${cycle.practice ?? "…"}\n\n### Feedback\n\n${cycle.feedback ?? "…"}`;
-    } catch (e) {
-      // Invalid schema
+      const explanation =
+        sectionToMarkdown("explanation", cycle.explanation ?? "…") ?? "…";
+      return `${content}\n\n### Explanation\n\n${explanation}\n\n### Check for understanding\n\n${cycle.checkForUnderstanding ? organiseAnswersAndDistractors(cycle.checkForUnderstanding) : "…"}\n\n### Practice\n\n${cycle.practice ?? "…"}\n\n### Feedback\n\n${cycle.feedback ?? "…"}`;
```

Comment:
I might revert this because it's still baffling!

Reply:
whoa!

Reply:
I've refactored this because it had a code complexity warning. Verified the output is the same